app/channel.go

```diff
@@ -77,7 +77,13 @@ func (a *App) JoinDefaultChannels(teamId string, user *model.User, shouldBeAdmin
 	for _, channelName := range a.DefaultChannelNames() {
 		channel, channelErr := a.Srv().Store.Channel().GetByName(teamId, channelName, true)
 		if channelErr != nil {
-			err = channelErr
+			var nfErr *store.ErrNotFound
+			switch {
+			case errors.As(err, &nfErr):
+				err = model.NewAppError("JoinDefaultChannels", store.MISSING_CHANNEL_ERROR, nil, nfErr.Error(), http.StatusNotFound)
```

Comment:
I think we can get rid of the `store.MISSING_CHANNEL_ERROR` altogether and just use the error key directly. It is anyways a bit misleading now because the error is an app layer error but it is in the store package. This would also need changing in 2 places in the api4 layer.

Reply:
Yep, I thought about that but I forgot to mention it. I thing this part is a bit messy because some error keys are in constant and other simply hardcoded.

Reply:
Yeah totally agree. It's good that we are fixing these now :)